src/System.Devices.Gpio/System.Devices.Gpio/I2cExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Devices.Gpio
+{
+    public static class I2cExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void Write(this I2cDevice device, params byte[] values)
+        {
+            device.Write(values);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void WriteInt8(this I2cDevice device, byte value)
+        {
+            device.Write(value);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void WriteInt16(this I2cDevice device, ushort value)
+        {
+            device.Write(BitConverter.GetBytes(value));
```

Comment:
Well, I understand your concern but in the link and example you mentioned the array is a parameter of the GetBytes method, so it is not actually the same kind of handy API I want to provide. Actually, if the user already has a byte array she can call directly the original Write method that takes an array as parameter. I could use a different private static byte array for each data type (8, 16, 32 and 64 bits) but it won't be thread safe.

Reply:
I was suggesting that I2cDevice.Write should take non-byte[] values such as int and uint and inside it could handle the conversion itself, if needed.